src/discover.ts

```diff
@@ -0,0 +1,39 @@
+import * as fs from "fs";
+import * as path from "node:path";
+import { TileDatabase } from "./db/tileDatabase";
+import { TileDataset } from "./types/app";
+import { Dict } from "./types/utils";
+
+export const discoverTileDatasets = (root: string): Dict<TileDataset> => {
+    // We expect to find tile databases in a single nesting of folders under root, where each folder has the dataset
+    // name, and each *.mbtiles files in the folder has the level name. These are the dataset and level identifiers
+    // which should be used in urls to access the tile data.
```

Comment:
It might help readers to say what a 'level' is in this context.

> [...] level (e.g. zoom level or administrative level) [...]

Or explain it in src/types/app.ts.